src/main/java/teammates/ui/controller/InstructorFeedbacksPageData.java

```diff
@@ -8,40 +9,325 @@
+                                                 || Const.TIME_REPRESENTS_FOLLOW_OPENING.equals(
+                                                           newFeedbackSession.sessionVisibleFromTime));
+        
+        newFsForm.setSessionVisiblePrivateChecked(newFeedbackSession != null 
+                                                  && Const.TIME_REPRESENTS_NEVER.equals(
+                                                           newFeedbackSession.sessionVisibleFromTime));
+                        
+        boolean hasResultVisibleDate = newFeedbackSession != null 
+                                       && !TimeHelper.isSpecialTime(newFeedbackSession.resultsVisibleFromTime);
+        
+        newFsForm.setResponseVisibleDateChecked(hasResultVisibleDate);
+        
+        newFsForm.setResponseVisibleDateValue(hasResultVisibleDate ? 
+                                              TimeHelper.formatDate(newFeedbackSession.resultsVisibleFromTime) :
+                                              "");
+        
+        newFsForm.setResponseVisibleDisabled(!hasResultVisibleDate);
+        
+        date = hasResultVisibleDate ? newFeedbackSession.resultsVisibleFromTime :  null;
+        
+        newFsForm.setResponseVisibleTimeOptions(getTimeOptionsAsElementTags(date));
+        
+        newFsForm.setResponseVisibleImmediatelyChecked((newFeedbackSession != null 
+                                                          && Const.TIME_REPRESENTS_FOLLOW_VISIBLE.equals(newFeedbackSession.resultsVisibleFromTime)));
+        
+        newFsForm.setResponseVisiblePublishManuallyChecked(
+                                 (newFeedbackSession == null 
+                                  || Const.TIME_REPRESENTS_LATER.equals(newFeedbackSession.resultsVisibleFromTime) 
+                                  || Const.TIME_REPRESENTS_NOW.equals(newFeedbackSession.resultsVisibleFromTime)));
+        
+        newFsForm.setResponseVisibleNeverChecked((newFeedbackSession != null  
+                                                    && Const.TIME_REPRESENTS_NEVER.equals(newFeedbackSession.resultsVisibleFromTime)));
+                                
+        newFsForm.setSubmitButtonDisabled(courses.isEmpty());
+    }
+    
+    
+    private List<FeedbackSessionRow> convertFeedbackSessionAttributesToSessionRows(
+                                    List<FeedbackSessionAttributes> sessions, 
+                                    HashMap<String, InstructorAttributes> instructors, 
+                                    String feedbackSessionNameForSessionList, String courseIdForNewSession) {
+
+        
+        List<FeedbackSessionRow> rows = new ArrayList<FeedbackSessionRow>();
+        int displayedStatsCount = 0;
+        
+        for (FeedbackSessionAttributes session : sessions) {
+            String courseId = session.courseId;
+            String name = PageData.sanitizeForHtml(session.feedbackSessionName);
```

Comment:
Considering this inherits from `PageData`, there's actually no need to use a class reference, is there?